### PR TITLE
Link relevant demo for classical shadows in docs

### PR DESCRIPTION
**Pull request by Qottmann**
Quick PR linking the demo on classical shadow in the module description and class docs string. The comment about preferably using qubit-wise-commuting groups instead is already made, now providing further details with our relevant demo.

### CODE REVIEW

File: pennylane/shadows/classical_shadow.py [hunk 1]
```diff
@@ -53,7 +54,7 @@ class ClassicalShadow:
             they appear in the columns of ``bits`` and ``recipes``. If None, defaults
             to ``range(n)``, where ``n`` is the number of measured wires.
 
-    .. seealso:: `PennyLane demo on Classical Shadows <https://pennylane.ai/qml/demos/tutorial_classical_shadows.html>`_, :func:`~.pennylane.classical_shadow`
+    .. seealso:: `Estimating observables with classical shadows in the Pauli basis <https://pennylane.ai/qml/demos/tutorial_diffable_shadows.html>`_, `PennyLane demo on Classical Shadows <https://pennylane.ai/qml/demos/tutorial_classical_shadows.html>`_, :func:`~.pennylane.classical_shadow`
```
Comment (dwierichs) [inline pennylane/shadows/classical_shadow.py:57]: Just thinking: Does the order matter? :thinking:

Comment (dwierichs) [reply pennylane/shadows/classical_shadow.py:57]: And should the word "demo" appear in the first link text as well?

File: pennylane/shadows/__init__.py [hunk 2]
```diff
@@ -44,7 +44,8 @@
 ---------------------------
 
 .. note:: As per `arXiv:2103.07510 <https://arxiv.org/abs/2103.07510>`_, when computing multiple expectation values it is advisable to directly estimate the desired observables by simultaneously measuring
-    qubit-wise-commuting terms. One way of doing this in PennyLane is via :class:`~pennylane.Hamiltonian` and setting ``grouping_type="qwc"``.
+    qubit-wise-commuting terms. One way of doing this in PennyLane is via :class:`~pennylane.Hamiltonian` and setting ``grouping_type="qwc"``. For more details on this topic, see our demo
```
Comment (dwierichs) [inline pennylane/shadows/__init__.py:47]:
```suggestion
    qubit-wise-commuting terms. One way of doing this in PennyLane is via :class:`~pennylane.Hamiltonian` and setting ``grouping_type="qwc"``. For more details on this topic, see the PennyLane demo
```

**Review by dwierichs — APPROVED** [review]
Thanks for taking care of this @Qottmann ! :+1: 
Just one rephrasing suggestion, but not blocking.

**Review by mudit2812 — APPROVED** [review]
Looks good @Qottmann ! I like David's suggestions as well 🚀

File: pennylane/shadows/classical_shadow.py (hunk 1 above)
Comment (Qottmann) [reply pennylane/shadows/classical_shadow.py:57]: I was even considering removing the second demo 😅

**Pull request merged**